components/plan-build/bin/shared.bash

```diff
@@ -132,12 +141,12 @@ _render_metadata_LD_RUN_PATH() {
     local metadata_file_name="LD_RUN_PATH"
 
     local ld_run_path_part=()
-    for lib in ${pkg_lib_dirs[@]}; do
+    for lib in "${pkg_lib_dirs[@]}"; do
         ld_run_path_part+=("${pkg_prefix}/$lib")
     done
-    if [[ -n ${ld_run_path_part} ]]; then
+    if [[ -n ${ld_run_path_part[*]} ]]; then
```

Comment:
Similar to https://github.com/habitat-sh/habitat/pull/4974#discussion_r192926905